tests/unit/nupic/encoders/date_test.py

```diff
@@ -166,6 +169,31 @@ def testWeekend(self):
       else:
         self.assertFalse(d.weekday()==0)
 
+
+  def testReadWrite(self):
+    originalTS = datetime.datetime(1997, 8, 29, 2, 14)
+    originalValue = self._e.encode(originalTS)
+
+    proto1 = DateEncoderProto.new_message()
+    self._e.write(proto1)
+
+    # Write the proto to a temp file and read it back into a new proto
+    with tempfile.TemporaryFile() as f:
+      proto1.write(f)
+      f.seek(0)
+      proto2 = DateEncoderProto.read(f)
+
+    encoder = DateEncoder.read(proto2)
+
+    self.assertIsInstance(encoder, DateEncoder)
+    self.assertTrue(numpy.array_equal(self._bits, encoder.encode(self._d)))
```

Comment:
Can you check a few of the attributes like `width` and maybe that the same set of encoders are set?